src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4161,9 +4161,10 @@ public SearchResult(final int index, final Rope rope) {
         @Specialization(guards = "isRubiniusByteArray(bytes)")
         public DynamicObject stringFromByteArray(DynamicObject bytes, int start, int count,
                                                  @Cached("create()") StringNodes.MakeStringNode makeStringNode) {
-            // Data is copied here - can we do something COW?
             final ByteArrayBuilder builder = Layouts.BYTE_ARRAY.getBytes(bytes);
-            return makeStringNode.fromRope(RopeBuilder.createRopeBuilder(builder, start, count).toRope(CR_UNKNOWN));
```

Comment:
Maybe we should remove this toRope() method and use a MakeStringNode instead since it goes slow path when we already have a nice `byte[]` around the corner.

Reply:
Indeed. I hadn't realized `toRope` was slow-path. I'll fix fast-path usages in another PR.